swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -393,6 +420,17 @@ private class DictionarySubscriptNode extends NodeImpl, TDictionarySubscriptNode
   override Location getLocationImpl() { result = expr.getLocation() }
 
   SubscriptExpr getExpr() { result = expr }
+
+  override DataFlowType getTypeImpl() {
+    exists(BoundGenericType dictType, TupleType tupleType |
+      dictType = expr.getBase().getType().getCanonicalType() and
+      tupleType.getType(0) = dictType.getArgType(0) and
+      tupleType.getType(1) = dictType.getArgType(1) and
+      tupleType.getNumberOfTypes() = 2 and
+      result.asType() = tupleType
+      // TODO: handle generic parameter types
+    )
+  }
```

Comment:
This looks very complicated 😨. Since `DictionarySubscriptNode` is really just an intermediate node that follows a subscript so that we can push both a `TupleContent` and a `CollectionContent`: could we instead simply get the type here by forwarding the call to the node representing the subscript expression?